### DISCUSSION TITLE
Simultaneously decode multiple mp3 files

### DISCUSSION THREAD

**Discussion by drfhaust** [earlephilhower/ESP8266Audio#617] [Q&A]
Is there a way to decode multiple mp3 files simultaneously with this library  and output the sum of the mp3 to an i2s device

**Comment by artificiel** [comment]
here is an example mixer with WAV from PROGMEM; you can adapt it to use MP3 Generator from whichever source you are using.

https://github.com/earlephilhower/ESP8266Audio/blob/master/examples/MixerSample/MixerSample.ino

starting a stream takes a bit of CPU and generally there's a hick-up when starting a MP3 while another one is already in the process of being decoded (ESP32 Wrover; freertos with a core dedicated to high-priority audio task). the hiccup can be mitigated by skipping ID3 and tweaking the buffer size but i did not manage total cleanliness. 

i managed to cleanly mix WAV, which have higher bandwidth but no processing. maybe the MP3 generator can be improved? (perhaps by splitting the MP3 setup over a few steps that allow the task to loop other generators while the decoder is being organized?). but it's a bit of work.
